doc/compiler/osr/OSR.md

```diff
@@ -0,0 +1,237 @@
+                 \            |
+                  \           |
+                    --------------------------
+                    |potentialOSRPointHelper1|---> catch 1
+                    --------------------------
+                              |
+                    -------------------------     ---------------------    ---------------------
+                    |     exit              |     | catch 2           |    | catch 1           |
+                    -------------------------     ---------------------    ---------------------
+                                                           |                        |
+                                                  ---------------------    ---------------------
+                                                  |prepareForOSRHelper|___>|prepareForOSRHelper|__> exit
+                                                  |   iconst 0        |    |  iconst -1        |
+                                                  ---------------------    ---------------------
+After OSRGuardinsertion:
+
+                    --------------------------
+                    |     entry              |---> catch 1
+                    --------------------------
+                             |
+                    --------------------------
+                    |   OSR guard            |
+                    --------------------------
+                   /         |
+                  /          |
+  ----------------  --------------------------
+  |jitInduceOSR  |  |  non-overriden guard 1 |---> catch 1
+  |AtCurrentPC   |  |                        |
+  ----------------  --------------------------
+        |          /         |
+    catch 1       /          |
+                 /           |
+  ---------------   --------------------------
+  |call B(cold) |   |    call C (from B)     |---> catch 2
+  ---------------   --------------------------
+       |         \           |
+    catch 1       \          |
+                    --------------------------
+                    |                        |---> catch 1
+                    --------------------------
+                             |
+                    --------------------------    ---------------------    ---------------------
+                    |     exit               |    | catch 2           |    | catch 1           |
+                    --------------------------    ---------------------    ---------------------
+                                                          |                         |
+                                                  ---------------------    ---------------------
+                                                  |prepareForOSRHelper|___>|prepareForOSRHElper|___>  exit
+                                                  |   iconst 0        |    |  iconst -1        |
+                                                  ---------------------    ---------------------
+
+```
```

Comment:
After this we'll also do OSR exception edge removal, which removes all OSR exception edges except those that originate from OSR induce blocks

I think it would be good to show this example CFG after that transformation as well. In particular, the OSR catch block and OSR code block for B will become unreachable and get removed

Reply:
Right. Just got tired of drawing :) Will add another CFG.

Reply:
done. Please take a look.